calculator.js

```diff
@@ -0,0 +1,185 @@
+let state = {
+  currentOperation: null, // это может быть сложение, умножение, вычитание, etc. Ключи: -, + и тд
+  immediateResult: 0, // "мгновенный" результат. Используется только для рендеринга, !не для расчетов!
+  fixedResult: 0, // используется как опорное значение. На его основе считается immediateResult
+  currentOperand: "", // это буфер, хранящий текущее число. Чтоб не плюсовали 1, потом 2, потом 3, а сразу 123
+};
+
+const setState = (key, payload) => {
+  state = { ...state, [key]: payload };
+  render();
+  console.log(state);
+};
+
+function setInitialState() {
+  setState("currentOperation", null);
+  setState("immediateResult", 0);
+  setState("fixedResult", 0);
+  setState("currentOperand", "");
+}
+
+function render() {
+  const { currentOperand, currentOperation, fixedResult, immediateResult } =
+    state;
+  const isAllClearButtonClicked =
+    currentOperand === "" &&
+    currentOperation === null &&
+    fixedResult === 0 &&
+    immediateResult === 0;
+
+  const isEqualClicked =
+    currentOperand === "" && currentOperation === null && fixedResult !== 0;
+
+  if (isAllClearButtonClicked) {
+    INPUT.value = "";
+    RESULT_SCREEN.innerHTML = 0;
+    SCREEN_EQUAL.classList.remove("active");
+
+    return;
+  }
+  if (isEqualClicked) {
+    SCREEN_EQUAL.classList.add("active");
+    INPUT.value = "";
+    return;
+  }
+  RESULT_SCREEN.innerHTML = state.immediateResult;
+
+  const maxLength = 12;
+  if (RESULT_SCREEN.textContent.length > maxLength) {
+    RESULT_SCREEN.innerHTML = state.immediateResult;
+    RESULT_SCREEN.textContent =
+      RESULT_SCREEN.textContent.slice(0, maxLength) + "…";
+  }
```

Comment:
not optimized, innerHTML of RESULT_SCREEN is rendered twice

I think it's better to check length at the beginning, and after that we can assign value to innerHTML only once